mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -211,14 +213,17 @@ private object IDFModel {
     val n = v.size
     v match {
       case SparseVector(size, indices, values) =>
+        val newElements = new ArrayBuffer[(Int, Double)]
         val nnz = indices.size
-        val newValues = new Array[Double](nnz)
         var k = 0
         while (k < nnz) {
-          newValues(k) = values(k) * idf(indices(k))
+          val newValue = values(k) * idf(indices(k))
```

Comment:
This change doesn't affect the `idf` vector -- it's still always dense. The existing `transform` code was producing `SparseVector`s containing explicit zeros. This change removes the explicit zeros, but only when the input to `transform` (not the `idf`) is a `SparseVector`.